File: rust/src/models/minimum_commitment_input.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::minimum_commitment_input_scope::MinimumCommitmentInputScope;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct MinimumCommitmentInput {
    /// Decimal string in the plan currency.
    pub amount: String,

    pub scope: MinimumCommitmentInputScope,
}

impl MinimumCommitmentInput {
    pub fn new(amount: String, scope: MinimumCommitmentInputScope) -> Self {
        Self { amount, scope }
    }
}

File: typescript/src/models/minimumCommitmentInput.ts
// this file is @generated
import {
  type MinimumCommitmentInputScope,
  MinimumCommitmentInputScopeSerializer,
} from "./minimumCommitmentInputScope";

export interface MinimumCommitmentInput {
  /** Decimal string in the plan currency. */
  amount: string;

  scope: MinimumCommitmentInputScope;
}

export const MinimumCommitmentInputSerializer = {
  _fromJsonObject(object: any): MinimumCommitmentInput {
    return {
      amount: object["amount"],
      scope: MinimumCommitmentInputScopeSerializer._fromJsonObject(object["scope"]),
    };
  },

  _toJsonObject(self: MinimumCommitmentInput): any {
    return {
      amount: self.amount,
      scope: MinimumCommitmentInputScopeSerializer._toJsonObject(self.scope),
    };
  },
};
